pkg/libhssh/filter_test.go

```diff
@@ -0,0 +1,118 @@
+package libhssh
+
+import (
+	"fmt"
+	"regexp"
+	"testing"
+
+	cr "github.com/squarescale/cloudresolver"
+	"github.com/stretchr/testify/suite"
+)
+
+type FilterTestSuite struct {
+	suite.Suite
+}
+
+func (s *FilterTestSuite) TestNewFilterFromString() {
+	reg, err := regexp.Compile("foo")
+	s.Nil(err)
+
+	testCases := []struct {
+		desc           string
+		filter         string
+		expectError    bool
+		expectedFilter *Filter
+	}{
+		{
+			desc:           "Missing ':'",
+			filter:         "x",
+			expectError:    true,
+			expectedFilter: nil,
+		},
+		{
+			desc:           "Too much ':'",
+			filter:         "x:y:z",
+			expectError:    true,
+			expectedFilter: nil,
+		},
+		{
+			desc:           "Unmatchable field name && valid value regexp",
+			filter:         "xxx:valid",
+			expectError:    true,
+			expectedFilter: nil,
+		},
+		{
+			desc:           "Matchable field name && valid value regexp",
```

Comment:
(nit) shouldn't it be Invalid value regexp ???

Reply:
doh !